### DISCUSSION TITLE
Move multiple issues between repositories

### DISCUSSION THREAD

**Discussion by ptmkenny** [community/community#9695] [Projects and Issues]
As also requested [in the old github issue tracker](https://github.com/isaacs/github/issues/1605), it would be great to have a way to move multiple issues in one repo to another.

Example use case:

1. Merge repo A into repo B.
2. Move all issues of repo A into repo B in one operation. (Currently, this has to be done one-by-one.)

**Comment by emkaminsk** [comment]
Agreed this would be really good to have; also to move multiple issues from project A to project B as there is a limit on number of issues in one project (1200).

**Comment by JasonLovesDoggo** [comment]
bump

**Comment by HunteRoi** [comment]
Any news on this?

**Comment by fgnievinski** [comment]
a workaround using CLI:
https://jloh.co/posts/bulk-migrate-issues-github-cli/

**Reply by vincerubinetti** [reply]
I also found this, but it says it doesn't transfer labels:
https://github.com/arunnalla/bulk-transfer-github-issues

~Anyone know if the CLI will preserve labels?~ Just tried it, it doesn't seem to preserve labels. I guess this is a limitation with GitHub programmatic issue transferring. Come to think of it, I'm not sure manually transferring issues preserves labels.

**Reply by jpswade** [reply]
> This does not scrub any issue content. Content such as text references to other issues, pull requests, projects and teams will still appear in the description or comments. Labels will be transferred.

**Reply by PatMyron** [reply]
my understanding is that issue labels are preserved [iff](https://wikipedia.org/wiki/If_and_only_if) those labels also exist in the other repo

**Reply by ThiloteE** [reply]
> I also found this, but it says it doesn't transfer labels: https://github.com/arunnalla/bulk-transfer-github-issues
> 
link now leads to 404

**Comment by opegamanalitarok** [comment]
At this time, GitHub does not have a built-in feature to move multiple issues from one repository to another in a single operation. However, there are a few workarounds that may be helpful.

One option is to use a third-party tool like the GitHub Issue Mover, which allows you to move multiple issues at once between repositories. Another option is to use the GitHub API to automate the process of moving issues using a script or tool.

Keep in mind that both of these options require some technical expertise and may have limitations depending on your specific use case. It's always important to test any changes thoroughly and ensure that you have backups of your data before making any significant modifications.

Hope this helps! 🙌🙌

**Comment by joaomas** [comment]
Would another solution then be to make a full backup of the repository?

**Comment by dwolrdcojp** [comment]
Move all your repositories to a monorepo they said... it will be fun they said!

**Comment by dharma-code-chris** [comment]
I just found this whilst searching for similar issues before creating the same post myself. It would be a very helpful feature.

**Comment by mbbroberg** [comment]
Still difficult today 💀

**Comment by OlehPendrakovskyi** [comment]
bump

**Comment by arielshaqed** [comment]
This feature would greatly help us, too, in both private and public repositories.

**Comment by bisdevres** [comment]
easily done by a script like this:

```powershell
cd /location/where/you/cloned/the/source-repo
$issues = (gh issue list --state all --json number --limit 1000) | ConvertFrom-Json

foreach($issue in $issues){
   gh issue transfer $issue.number <owner>/<target-repo>
}
```

> [!warning]  
> be sure the target repo contains all necessary labels.  
> label which are not present there will be removed from your issues.

> [!note]  
> this required [GitHub CLI](https://cli.github.com/) to be installed.

**Comment by Neustradamus** [comment]
Dear @github @community team,

This ticket is really important.

Thanks in advance.

It is linked to:
- https://github.com/orgs/community/discussions/5766

**Comment by judovana** [comment]
Note, that the proper solution will allow also copy of the issues in case of organization change. Then the simple transfer is getting slightly complicated.

bump

**Comment by wh34t86** [comment]
Bump
